src/test/java/seedu/address/logic/parser/AddCommandParserTest.java

```diff
@@ -271,45 +271,49 @@ public void parse_invalidValue_failure() {
         // invalid name
         assertParseFailure(
                 parser, INVALID_NAME_DESC + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
-                        + TAG_DESC_HUSBAND + TAG_DESC_FRIEND + DATEOFLASTVISIT_DESC_BOB,
+                        + TAG_DESC_CONVICT + TAG_DESC_FRIEND + DATEOFLASTVISIT_DESC_BOB,
                 Name.MESSAGE_CONSTRAINTS);
 
         // invalid phone
         assertParseFailure(
                 parser, NAME_DESC_BOB + INVALID_PHONE_DESC + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
-                        + TAG_DESC_HUSBAND + TAG_DESC_FRIEND + DATEOFLASTVISIT_DESC_BOB,
+                        + TAG_DESC_CONVICT + TAG_DESC_FRIEND + DATEOFLASTVISIT_DESC_BOB,
                 Phone.MESSAGE_CONSTRAINTS);
 
         // invalid email
         assertParseFailure(
                 parser, NAME_DESC_BOB + PHONE_DESC_BOB + INVALID_EMAIL_DESC + ADDRESS_DESC_BOB
-                        + TAG_DESC_HUSBAND + TAG_DESC_FRIEND + DATEOFLASTVISIT_DESC_BOB,
+                        + TAG_DESC_CONVICT + TAG_DESC_FRIEND + DATEOFLASTVISIT_DESC_BOB,
                 Email.MESSAGE_CONSTRAINTS);
 
-        // invalid tag
+        // single invalid tag
         assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
-                + INVALID_TAG_DESC + VALID_TAG_FRIEND + DATEOFLASTVISIT_DESC_BOB, Tag.MESSAGE_CONSTRAINTS);
+                + INVALID_TAG_DESC + DATEOFLASTVISIT_DESC_BOB, Tag.MESSAGE_CONSTRAINTS);
+
+        // valid tag followed by invalid tag
+        assertParseFailure(parser, NAME_DESC_BOB + PHONE_DESC_BOB + EMAIL_DESC_BOB + ADDRESS_DESC_BOB
+                + TAG_DESC_CONVICT + INVALID_TAG_DESC + DATEOFLASTVISIT_DESC_BOB, Tag.MESSAGE_CONSTRAINTS);
 
```

Comment:
suggestion: might want to include invalid by valid then it would assume to cover all permutations? I'm not sure, do you think it is necessary?

Reply:
Good idea, I'll make some more permutations of this for clarity and exhaustiveness of testing.

I do foresee one unexpected behaviour with this, which is a command like `edit 1 t/friends colleague` that's intended to set 2 tags might be successfully parsed as a single tag of `friends colleague` instead of throwing an error, but I don't think there's any way around it and we can stipulate that as incorrect usage